spotify/rootfs/etc/services.d/spotifyd/run

```diff
@@ -14,6 +14,12 @@ bashio::log.info 'Starting the Spotify daemon...'
 # Bitrate
 options+=(--bitrate $(bashio::config 'bitrate'))
 
+# Initial volume
+options+=(--initial-volume $(bashio::config 'initial_volume'))
```

Comment:
I haven't tested these yet, but do they control the pulse audio channel, or does this apply internally to its own little magic?

Reply:
Good question - haven't checked how it's internally implemented. At a quick glance ([code here](https://github.com/librespot-org/librespot/tree/4925adb4f103cea20f38eee1db9f09b0d3e07e75/playback/src/mixer)) it looks like they use the Alsa mixer when Alsa is the backend, and a softmixer in all the other cases.

Reply:
The reason I ask, is that from the HA end, one of the goals is to provide controls for these things.

Reply:
Great idea!

`librespot` as far as I know, doesn't provide any way to be controlled remotely, except via the Spotify client.
There's a few options that might do the trick in the future:
- implementing a pulseaudio mixer in `librespot` (will only make volume changes possible - no actions for media control)
- using [spotcontrol](https://github.com/badfortrains/spotcontrol), but that will have a few issues (there's authentication involved, so with multiple Spotify users it will be cumbersome)
- implementing some kind of RPC in `librespot`, or even implementing a proper MPRIS interface. There was [a similar request](https://github.com/plietar/librespot/issues/100) long ago, but it only resulted in an interface to run scripts on events (which I currently use to [report](https://github.com/giannello/addon-spotify-connect-librespot/blob/master/spotify-librespot/rootfs/usr/local/bin/ha.sh) the librespot status to an HA entity.

The latter would be the most standard option looking forward. MPRIS integrations for HA are already [out there](https://github.com/rytilahti/homeassistant-mpris-bridge)

Reply:
Yeah, but you'll end up with 2 volume controls - one controlled by HA, one controlled by the spotify clients (librespot soft mixer). 2 independent volume controls for the same source are going to be an issue - if you want to have _coherent_ volume control via pulseaudio, then both HA and librespot should be able to do it via pulseaudio, otherwise - for example - lowering the volume via HA won't be reflected in the Spotify app.

Reply:
In that case, this should not be implemted at all, and the addon adjusted back to Alsa